smithy-model/src/main/java/software/amazon/smithy/model/transform/ModelTransformer.java

```diff
@@ -132,6 +136,38 @@ public Model removeShapesIf(Model model, Predicate<Shape> predicate) {
         return filterShapes(model, FunctionalUtils.not(predicate));
     }
 
+    /**
+     *  Renames shapes using ShapeId pairs while ensuring that the
+     *  transformed model is in a consistent state.
+     *
+     *  <p>This transformer ensures that when an aggregate shape is renamed, all
+     *  members are updated in the model.
+     *
+     * @param model Model to transform.
+     * @param renamed Map of shapeIds
+     * @return Returns the transformed model.base.
+     */
+    public Model renameShapes(Model model, Map<ShapeId, ShapeId> renamed) {
+        return new RenameShapes(renamed).transform(this, model);
```

Comment:
I suggest that you call the overloaded renameShapes method and pass in a default version of the supplier (e.g., `Model::assembler`).